platform/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/command/ProfileListCommandTest.java

```diff
@@ -0,0 +1,129 @@
+        ddfHome.newFolder("etc", "admin", "profiles");
+        profilePath = Paths.get(ddfHome.getRoot().toString(), "etc", "admin", "profiles");
+        Files.copy(this.getClass().getResourceAsStream("/profiles/devProfile.json"), Paths.get(profilePath.toAbsolutePath().toString(), "devProfile.json"));
+        Files.copy(this.getClass().getResourceAsStream("/profiles/profileWithDuplicates.json"), Paths.get(profilePath.toAbsolutePath().toString(), "profileWithDuplicates.json"));
+        out = new ByteArrayOutputStream();
+        console = new PrintStream(out);
+        profileListCommand = new ProfileListCommand();
+        profileListCommand.console = console;
+        profileListCommand.profilePath = profilePath;
+
+        this.applicationService = mock(ApplicationServiceImpl.class);
+        this.featuresService = mock(FeaturesServiceImpl.class);
+        this.bundleService = mock(BundleServiceImpl.class);
+
+        Feature feature = createMockFeature("standard");
+        List<Dependency> deps = Arrays.asList(createMockDependency("app1"),
+                createMockDependency("app2"), createMockDependency("app3"));
+        when(feature.getDependencies()).thenReturn(deps);
+        when(applicationService.getInstallationProfiles()).thenReturn(Collections.singletonList(feature));
+
+    }
+
+    @Test
+    public void testProfileFilesExist() {
+        profileListCommand.doExecute(applicationService, featuresService, bundleService);
+        assert out.toString().contains("standard");
+        assert out.toString().contains("devProfile");
+        assert out.toString().contains("profileWithDuplicates");
+    }
+
+    @Test
+    public void testProfileFilesNotExist() {
+        profileListCommand.profilePath = Paths.get("foo");
+        profileListCommand.doExecute(applicationService, featuresService, bundleService);
+        assert out.toString().contains("standard");
+        assert !out.toString().contains("devProfile");
+        assert !out.toString().contains("profileWithDuplicates");
+    }
+
+    @Test
+    public void testRegularProfileNotExistAndProfileFilesExist() {
+        when(applicationService.getInstallationProfiles()).thenReturn(Collections.emptyList());
+        profileListCommand.doExecute(applicationService, featuresService, bundleService);
+        assert !out.toString().contains("standard");
+        assert out.toString().contains("devProfile");
+        assert out.toString().contains("profileWithDuplicates");
+    }
+
+    @Test
+    public void testNoProfilesExist() {
+        when(applicationService.getInstallationProfiles()).thenReturn(Collections.emptyList());
+        profileListCommand.profilePath = Paths.get("foo");
+        profileListCommand.doExecute(applicationService, featuresService, bundleService);
+        assert !out.toString().contains("standard");
+        assert !out.toString().contains("devProfile");
+        assert !out.toString().contains("profileWithDuplicates");
+    }
+
```

Comment:
❗️ We need to test that the `IOException` is properly handled when `Files.list(profilePath)` is called in `listProfileFiles()`.